### PR TITLE
respect show.legend in colorbar

### LINKED ISSUE #1568: Omit legend using `show.legend = FALSE` does not work when aesthetic is continuous
`show.legend = FALSE` works as expected when the fill variable is discrete. However, when the `aes`thetic is mapped to a continuous variable, `show.legend = FALSE` does not drop the legend.

    df <- data.frame(x = 1:3, y = 1)

Discrete `fill` or `color` `aes`:
    
    ggplot(df, aes(x = x, y = y, color = factor(x))) +
      geom_point(size = 20, shape = 21, show.legend = FALSE)
    
    ggplot(df, aes(x = x, y = y, fill = factor(x))) +
      geom_point(size = 20, shape = 21, show.legend = FALSE)

Continuous `fill` or `color` `aes`:

    ggplot(df, aes(x = x, y = y, color = x)) +
      geom_point(size = 20, shape = 21, show.legend = FALSE)

    ggplot(df, aes(x = x, y = y, fill = x)) +
      geom_point(size = 20, shape = 21, show.legend = FALSE)

`ggplot2_2.1.0`

### PR DESCRIPTION
**Pull request by thomasp85**
Fixes #1568

### CODE REVIEW

File: R/guides-.r [hunk 1]
```diff
@@ -208,6 +208,11 @@ guides_merge <- function(gdefs) {
 
 # process layer information
 guides_geom <- function(gdefs, layers, default_mapping) {
+  # remove layers with show.legend=FALSE or NA
```
Comment (hadley) [inline R/guides-.r]: Are you sure this logic is correct? `NA` means use the default, i.e. show if the aesthetic is used

Comment (thomasp85) [inline R/guides-.r]: Oh - I'll fix

**Comment by hadley** [comment]
LGTM

File: R/guide-colorbar.r [hunk 2]
```diff
@@ -211,7 +211,22 @@ guide_merge.colorbar <- function(guide, new_guide) {
 
 # this guide is not geom-based.
 #' @export
-guide_geom.colorbar <- function(guide, ...) {
+guide_geom.colorbar <- function(guide, layers, defaul_mapping) {
+  # Layers that use this guide
+  guide_layers <- plyr::llply(layers, function(layer) {
+    matched <- matched_aes(layer, guide, defaul_mapping)
```
Comment (hadley) [inline R/guide-colorbar.r]: `defaul_mapping` not `default_mapping`

**Pull request merged**